### DISCUSSION TITLE
pake和pake-cli2.0主要变化 | Major changes in pake and pake-cli2.0

### DISCUSSION THREAD

**Discussion by Tlntin** [tw93/Pake#396] [Show and tell]
## pake和pake-cli2.0主要变化。

- 注意：目前还在alpha状态，可能会有Bug。

### pake-cli 2.0主要变化

- 增加了网络检测功能，根据用户当前网络ping外网的延迟，来决定是否自动换中国国内源，例如自动换npm淘宝源，自动切换字节跳动的rust源。
- 增加了打包本地静态文件的功能，直接将之前的url换成目标静态文件路径就可以了，并且可以用`--copy-iter-file`参数实现递归拷贝目标路径下面的所有文件（当然，单个静态文件就没必要用这个参数了）。
- 增加了自定义user-agent功能。
- 增加了系统托盘显示功能，通过`--show-system-tray`开启。
- 增加了菜单栏图标显示功能，通过`--show-menu`开启，推荐MacOS用户打开。

### pake 2.0主要变化

- 框架从wry切换到tauri，理论上更加稳定。
- 支持记住上次关闭应用的位置。
- 关闭按钮变成了最小化。
- 支持系统托盘。
- [后续可能]支持安卓，甚至ios应用，等上游tauri2.0稳定再说。

## Major changes in pake and pake-cli2.0.

- Note: Currently in alpha state, there may be bugs.

### main changes in pake-cli 2.0

- The network detection function has been added to determine whether to automatically switch to China's domestic source according to the delay of the user's current network pinging the external network, such as automatically switching to the npm Taobao source, and automatically switching to the ByteDance rust source.
- Added the function of packaging local static files, just replace the previous url with the target static file path, and you can use the `--copy-iter-file` parameter to recursively copy all files under the target path (of course, There is no need to use this parameter for a single static file).
- Added custom user-agent function.
- Added system tray display function, enabled by `--show-system-tray`.
- Added menu bar icon display function, enabled by `--show-menu`, recommended for MacOS users.

### main changes in pake 2.0

- The framework is switched from wry to tauri, which is theoretically more stable.
- Support remembering the location where the app was last closed.
- Close button becomes minimized.
- Support system tray.
- [Follow-up may] support Android, and even ios applications, wait until the upstream tauri2.0 is stable.

**Comment by tw93** [comment]
已发布 2.0 版本
